src/cargo/sources/git/source.rs

```diff
@@ -122,8 +123,20 @@ impl<'cfg> Source for GitSource<'cfg> {
             return Ok(());
         }
 
-        let git_path = self.config.git_path();
-        let git_path = self.config.assert_package_cache_locked(&git_path);
+        let git_fs = self.config.git_path();
```

Comment:
Wonder why `CACHEDIR.TAG` of `GitSource` is implemented under `fn block_until_ready`, whereas `RegistrySource` is under `RegistrySource::remote`. I feel like we can put both `CACHEDIR.TAG` creation logic on "initialize" methods, such as `GitSource::new()`. What do you think?

Reply:
Done